sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs

```diff
@@ -171,11 +183,24 @@ private int CompareBodies(byte[] requestBody, byte[] responseBody, StringBuilder
             IsEquivalentUri(entry.RequestUri, otherEntry.RequestUri) &&
             CompareHeaderDictionaries(entry.Request.Headers, otherEntry.Request.Headers, VolatileHeaders) == 0;
 
-        private static bool AreUrisSame(string entryUri, string otherEntryUri) =>
-            // Some versions of .NET behave differently when calling new Uri("...")
-            // so we'll normalize the recordings (which may have been against
-            // a different .NET version) to be safe
-            new Uri(entryUri).ToString() == new Uri(otherEntryUri).ToString();
+        private bool AreUrisSame(string entryUri, string otherEntryUri) =>
+            TranslateUri(entryUri) == TranslateUri(otherEntryUri);
+
+        private string TranslateUri(string uriToTranslate)
+        {
+            var req = new RequestUriBuilder();
+            var uri = new Uri(uriToTranslate);
+            req.Reset(uri);
+            req.Query = "";
+            NameValueCollection queryParams = HttpUtility.ParseQueryString(uri.Query);
+            foreach (string param in queryParams)
+            {
+                req.AppendQuery(
+                    param,
+                    VolatileQueryParameters.Contains(param) ? VolatileValue : queryParams[param]);
+            }
+            return req.ToUri().ToString();
```

Comment:
We've stopped using `Uri.ToString()` in favor of `Uri.AbsoluteUri`.  It has fewer encoding problems.  Looks like we didn't update it in the `RecordMatcher`, but we probably should.

Reply:
Actually, looks like this change breaks some of the storage tests as the encoding for "[" changes between NET framework and core.

Reply:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=454987&view=ms.vss-test-web.build-test-results-tab&runId=11919522&resultId=115490&paneView=debug